ibis/backends/exasol/compiler.py

```diff
@@ -178,3 +177,6 @@ def visit_DateDelta(self, op, *, part, left, right):
         #       * part = hour -> hours_between
         #       * ...
         return self.f.days_between(left, right)
+
+    def visit_ExtractDayOfYear(self, op, *, arg):
+        return self.f.to_char(arg, "DDD")
```

Comment:
```suggestion
        return self.cast(self.f.to_char(arg, "DDD"), op.dtype)
```

So that the casting to int is done in the database and not in-memory by pandas/pyarrow.

Reply:
Thanks @cpcloud. Initially, I had it like that. The tests have been passing without it 😅. I must admit, I'm still getting acquainted with the new architecture/structure. However, I really like it 👍.

Reply:
Yep, they're passing because we cast results in python to ensure that the output dtypes match the ibis expression's types. In this case, it's much less efficient presumably, because casting from string to int in exasol is likely much faster than computing a bunch of strings only to throw them away immediately.